composer/trainer/trainer.py

```diff
@@ -1020,9 +1099,248 @@ def saved_checkpoints(self) -> List[Tuple[Timestamp, List[pathlib.Path]]]:
+
+            optimizers (torch.optim.Optimizer | Sequence[torch.optim.Optimizer], optional): See :class:`.Trainer`.
+            schedulers (PyTorchScheduler | ComposerScheduler | Sequence[PyTorchScheduler | ComposerScheduler], optional):
+                See :class:`.Trainer`.
+            scale_schedule_ratio (float, optional): See :class:`.Trainer`.
+            step_schedulers_every_batch (bool, optional): See :class:`.Trainer`.
+            eval_dataloader (Iterable | DataSpec | Evaluator | Sequence[Evaluator], optional): See :class:`.Trainer`.
+            eval_subset_num_batches (int, optional): See :class:`.Trainer`.
+            eval_interval (int | str | Time | (State, Event) -> bool, optional): See :class:`.Trainer`.
+            grad_accum (int | str, optional): See :class:`.Trainer`.
+            precision (Precision | str, optional): See :class:`.Trainer`.
+            grad_clip_norm (float, optional): See :class:`.Trainer`.
+        """
+        # Train Dataloader
+        if train_dataloader is not None:
+            self._train_data_spec = ensure_data_spec(train_dataloader)
+            self.state.set_dataloader(self._train_data_spec.dataloader, train_dataloader_label)
+        if self._train_data_spec is None:
+            _raise_missing_argument_exception("train_dataloader")
+        if train_subset_num_batches is not None:
+            self.state.dataloader_len = train_subset_num_batches
+        if compute_training_metrics is not None:
+            self.train_metrics = _get_training_metrics(self._original_model) if compute_training_metrics else None
+
+        # Reset Time
+        if reset_time:
+            self.state.timestamp = Timestamp()
+
+        # Max Duration
+        if duration is not None:
+            duration = ensure_time(duration, TimeUnit.EPOCH)
+            # Effectively increment the max duration (if not resetting the Time)
+            # or set the max_duration (if resetting the time -- self.state.timestamp.get(duration.unit) will be 0)
+            # It is important to set the duration, rather than incrementing it, as ``duration`` could be in
+            # different units than ``max_duration``
+            self.state.max_duration = duration + self.state.timestamp.get(duration.unit)
+
+        if self.state.max_duration is None:
+            _raise_missing_argument_exception("max_duration")
+
+        if self.state.max_duration <= self.state.timestamp.get(self.state.max_duration.unit) and not reset_time:
+            raise ValueError(
+                (f"The max_duration ({self.state.max_duration}) is less than or equal to the elapsed training duration "
+                 f"({self.state.timestamp.get(self.state.max_duration.unit)}). No training would occur. "
+                 "Please provide the `duration` or specify `reset_time=True` in Trainer.fit()."))
+
+        # Scale Schedule Ratio and Schedulers
+        if scale_schedule_ratio != 1.0:
```

Comment:
It might be simpler to call this function even if ssr is 1.0 because it just multiplies the value by 1.0.

Reply:
We don't attempt to scale the scheduler if the ratio is 1.0, in case if the scheduler cannot be scaled (e.g. the user passes in a custom PyTorch LambdaLR). I'll add a comment here to explain.

Reply:
Added the comment.